server/ServerFSM.cpp

```diff
@@ -1158,7 +1158,8 @@ sc::result MPLobby::react(const JoinGame& msg) {
         if (authenticated)
             player_connection->SetAuthenticated();
     } else {
-        if (client_type != Networking::ClientType::CLIENT_TYPE_AI_PLAYER && server.IsAuthRequiredOrFillRoles(player_name, roles)) {
+        const bool relaxed_auth = player_connection->IsLocalConnection() && client_type != Networking::ClientType::CLIENT_TYPE_AI_PLAYER;
```

Comment:
Should probably only relax if the player type is Observer. Relaxed authentication for a moderator would be more problematic than for a single empire player...

Reply:
> I don't understand this... Why the not-an-AI requirement? Would be good to comment to explain the logic / expectations...

Because AI could not answer passwords to authentication requests.

> And what does relaxed authentication mean, exactly? What is the alternative? (No authentication required, or no connecton possible?)

It means local AI can join the game without authentication check, no authentication required for it. Not sure what you mean for alternative